botfront/imports/api/importExport/export.methods.js

```diff
@@ -32,7 +53,329 @@ if (Meteor.isServer) {
+                await repo.mergeBranches(
+                    branchName,
+                    `refs/remotes/origin/${branchName}`,
+                    signature(),
+                    nodegit.Merge.PREFERENCE.FASTFORWARD_ONLY,
+                    { fileFavor: nodegit.Merge.FILE_FAVOR.THEIRS },
+                );
+                const localSha = (await repo.getBranchCommit(branchName)).id().tostrS();
+                const remoteSha = (
+                    await repo.getBranchCommit(`refs/remotes/origin/${branchName}`)
+                )
+                    .id()
+                    .tostrS();
+                // local is ahead, start from scratch
+                if (localSha !== remoteSha) throw new Error();
+            } catch {
+                if (fs.existsSync(dir)) fs.rmdirSync(dir, { recursive: true });
+                repo = await nodegit.Clone.clone(url, dir, opts);
+            }
+            branch = await repo.getBranch(branchName);
+            branchCommit = await repo.getBranchCommit(branchName);
+            remote = await repo.getRemote('origin');
+        } catch {
+            throw new Meteor.Error(
+                `Could not connect to branch '${branchName}' on your git repo. Check your credentials.`,
+            );
+        }
+        const index = await repo.index();
+        await index.read(1);
+
+        const bfIgnoreFile = join(dir, '.bfignore');
+        const exclusions = [
+            '.bfignore',
+            ...(fs.existsSync(bfIgnoreFile)
+                ? fs.readFileSync(bfIgnoreFile, 'utf-8').split('\n')
+                : []),
+        ];
+
+        return {
+            url,
+            branchName,
+            dir,
+            index,
+            repo,
+            branch,
+            branchCommit,
+            remote,
+            exclusions,
+            opts,
+        };
+    };
+
+    const repoDirToFileList = async (dir, exclusions) => glob.sync(join('**', '*'), { cwd: dir, nodir: true }).reduce(
+        (files, path) => [
+            ...files,
+            ...(exclusions.includes(path)
+                ? []
+                : [
+                    {
+                        createReadStream: () => fs.createReadStream(join(dir, path)),
+                        filename: basename(path),
+                    },
+                ]),
+        ],
+        [],
+    );
+
+    const commitChanges = async (projectId, commitMessage, { status, ...repoInfo }) => {
+        const nodegit = require('nodegit');
+        const {
+            dir, index, repo, branchCommit, exclusions,
+        } = repoInfo;
+        // given that the present method is called on training, calling 'exportRasa'
+        // here means 'rasa.getTrainingPayload' will be called twice in short succession,
+        // which is a bit stupid, but we live with it for now.
+        const zip = await Meteor.callWithPromise('exportRasa', projectId, 'all', {
+            noBlob: true,
+        });
```

Comment:
it does not export conversation or incoming with this call

Reply:
It was on purpose, since we are versioning a bot, not data from bot usage...

Reply:
damn I've made a commit for that